controllers/addresspool_controller.go

```diff
@@ -213,3 +230,58 @@ func (r *AddressPoolReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&starlingxv1.AddressPool{}).
 		Complete(r)
 }
+
+// Verify whether we have annotation restore-in-progress
+func (r *AddressPoolReconciler) checkRestoreInProgress(instance *starlingxv1.AddressPool) bool {
+	restoreInProgress, ok := instance.Annotations[cloudManager.RestoreInProgress]
+	if ok && restoreInProgress != "" {
+		return true
+	}
+	return false
+}
+
+// Update status
+func (r *AddressPoolReconciler) RestoreAddressPoolStatus(instance *starlingxv1.AddressPool) error {
+	annotation := instance.GetObjectMeta().GetAnnotations()
+	config, ok := annotation[cloudManager.RestoreInProgress]
+	if ok {
+		restoreStatus := &cloudManager.RestoreStatus{}
+		err := json.Unmarshal([]byte(config), &restoreStatus)
+		if err == nil {
```

Comment:
we should check the err first, log and return nil

Reply:
Implemented this way